autoload/test/javascript/lab.vim

```diff
@@ -0,0 +1,50 @@
+if !exists('g:test#javascript#lab#file_pattern')
+  let g:test#javascript#lab#file_pattern = '\vtest/.*\.js$'
+endif
+
+function! test#javascript#lab#test_file(file) abort
+  return a:file =~# g:test#javascript#lab#file_pattern
+endfunction
+
+function! test#javascript#lab#test_file(file) abort
+  if a:file =~# g:test#javascript#lab#file_pattern
+    if exists('g:test#javascript#runner')
+      return g:test#javascript#runner == 'lab'
+    else
+      return executable('lab')
```

Comment:
Here we could rather check `filereadable('node_modules/.bin/lab')`, if we decide we want to make declaring `lab` as a dependency in `package.json` mandatory.